modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -36,39 +42,64 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.stream.Collectors;
 
-public class Netty4HttpRequest implements HttpRequest {
+public class Netty4HttpRequest implements org.elasticsearch.http.HttpRequest {
 
-    private final FullHttpRequest request;
+    private final HttpRequest nettyRequest;
+    private final HttpContent nettyContent;
+    private final Netty4RequestContentPublisher contentPublisher;
     private final BytesReference content;
     private final Map<String, List<String>> headers;
     private final AtomicBoolean released;
     private final Exception inboundException;
     private final boolean pooled;
     private final int sequence;
 
+    Netty4HttpRequest(int sequence, HttpRequest request, Netty4RequestContentPublisher contentPublisher) {
+        this.sequence = sequence;
+        this.nettyRequest = request;
+        this.nettyContent = new DefaultHttpContent(Unpooled.EMPTY_BUFFER);
+        this.contentPublisher = contentPublisher;
+        this.content = BytesArray.EMPTY;
+        this.released = new AtomicBoolean(false);
+        this.pooled = false;
+        this.headers = getHttpHeadersAsMap(request.headers());
+        this.inboundException = null;
+    }
```

Comment:
Add support to Netty4HttpRequest to be HttpRequest and optional HttpContent, rather than always FullHttpRequest. I do drop trailing headers for not full requests, not sure do we use them or not. Haven't found traces of them yet.